src/datasource.ts

```diff
@@ -168,61 +168,70 @@ export default class CogniteDatasource {
       if (queryList.length === 0 || target.error) {
         continue;
       }
-      // keep track of target lengths so we can assign errors later
-      targetQueriesCount.push({
-        refId: target.refId,
-        count: queryList.length,
-      });
-      // create query requests
-      const queryReq: DataQueryRequest = {
-        items: queryList,
-        start: timeFrom,
-        end: timeTo,
-      };
-      if (target.aggregation && target.aggregation !== 'none') {
-        queryReq.aggregates = target.aggregation;
-        if (!target.granularity) {
-          queryReq.granularity = this.intervalToGranularity(options.intervalMs);
-        } else {
-          queryReq.granularity = target.granularity;
-        }
-      }
-      if (target.assetQuery && target.assetQuery.func && target.tab === Tab.Custom) {
-        if (queryReq.aggregates) {
-          target.error =
-            '[ERROR] To use aggregations with functions, use [ID,aggregation,granularity] or [ID,aggregation]';
-          targetQueriesCount.pop();
-          continue;
-        }
-        let ids = 0;
-        const idRegex = /\[.*?\]/g; // look for [something]
-        for (const q of queryList) {
-          const matches = q.function.match(idRegex);
-          if (!matches) break;
-          const idsObj = {};
-          for (const match of matches) {
-            idsObj[match.substr(1, match.length - 2)] = true;
+
+      // /dataquery is limited to 100 items, so we need to add new calls if we go over 100 items
+      while (queryList.length > 0) {
+        const slicedQl = queryList.slice(0, 100); // only get first 100 items
+
+        // keep track of target lengths so we can assign errors later
+        targetQueriesCount.push({
+          refId: target.refId,
+          count: slicedQl.length,
+        });
+        // create query requests
+        const queryReq: DataQueryRequest = {
+          items: slicedQl,
+          start: timeFrom,
+          end: timeTo,
+        };
+        if (target.aggregation && target.aggregation !== 'none') {
+          queryReq.aggregates = target.aggregation;
+          if (!target.granularity) {
+            queryReq.granularity = this.intervalToGranularity(options.intervalMs);
+          } else {
+            queryReq.granularity = target.granularity;
           }
-          ids += Object.keys(idsObj).length;
         }
-        if (ids === 0) ids = 1; // will fail anyways, just show the api error message
+        if (target.assetQuery && target.assetQuery.func && target.tab === Tab.Custom) {
+          if (queryReq.aggregates) {
+            target.error =
+              '[ERROR] To use aggregations with functions, use [ID,aggregation,granularity] or [ID,aggregation]';
+            break;
+          }
+          let ids = 0;
+          const idRegex = /\[.*?\]/g; // look for [something]
+          for (const q of slicedQl) {
+            const matches = q.function.match(idRegex);
+            if (!matches) break;
```

Comment:
Do you intentionally break here? Or do you want to `continue` to the next `q`.

Reply:
Each of these queries all belong to the same query so they would all have the same function. So if one of them doesn't have an id in its function, then none of them do